AY2324S2-CS2103T-F12-1/tp
# Refactor RelateStorage to be under NetConnect.json and Implement Unrelate Command

# Refactor Relationship Storage & Implement Unrelate Command

## Overview
This pull request contains changes that refactor the storage system of our application to incorporate relationship data within `NetConnect.json`. It also includes the implementation of the 'unrelate' command, which allows users to remove existing relationships between entities.

## Changes
* Refactored storage to consolidate all data into `NetConnect.json` for better data integrity and management.
* Added JSON objects for `IdTuple` to facilitate the new storage structure.
* Implemented the 'unrelate' command to remove relationships between entities.
* Updated dependencies to support the refactored storage and new command.
* Performed comprehensive updates to unit testing to cover new functionalities.

## Impact
* Simplifies data architecture by eliminating the need for a separate relations file.
* Enhances user control over relationship data with the ability to add and remove relationships as needed.
* Improves maintainability and scalability of the codebase.

## Testing
* All existing tests have been updated, and new tests have been added to ensure full coverage of the new features.
* Manual testing has been conducted to validate that the 'unrelate' command performs as expected.

## Discussion
* The 'unrelate' command syntax has been aligned with our existing command structure for consistency.
* Exception handling has been thoroughly reviewed to handle edge cases gracefully.

## Instructions for Reviewers
1. Review the updated storage structure in `NetConnect.json`.
2. Evaluate the new JSON objects for `IdTuple` and related changes in the `JsonSerializableNetConnect` class.
3. Examine the implementation of the 'unrelate' command in `UnrelateCommand.java`.
4. Assess the changes to the dependencies and their implications on the project.
5. Verify the updates to the unit tests and their coverage of the new features.

## Steps for Testing
* Run the updated unit tests to confirm they pass without issues.
* Check out the branch and manually test the 'unrelate' command in various scenarios.
* Verify that relationship data is correctly updated in the `NetConnect.json` file after using the 'unrelate' command.

## References
* [Refactor relate storage](https://github.com/AY2324S2-CS2103T-F12-1/tp/issues/109)
* [Unrelate command](https://github.com/AY2324S2-CS2103T-F12-1/tp/issues/112)
